tests/test-polynomial-matrix.C

```diff
@@ -0,0 +1,425 @@
+/*
+ * Copyright (C) 2013  Pascal Giorgi
+ *
+ * Written by Pascal Giorgi <pascal.giorgi@lirmm.fr>
+ *
+ * ========LICENCE========
+ * This file is part of the library LinBox.
+ *
+ * LinBox is free software: you can redistribute it and/or modify
+ * it under the terms of the  GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ * ========LICENCE========
+ */
+
+
+
+
+#include <iostream>
+#include <iomanip>
+using namespace std;
+
+#include <linbox/ring/modular.h>
+#include <linbox/randiter/random-prime.h>
+#include <linbox/randiter/random-fftprime.h>
+#include <givaro/zring.h>
+#include <recint/rint.h>
+#include <linbox/matrix/matrix-domain.h>
+#include <linbox/util/commentator.h>
+#include <linbox/util/timer.h>
+#include <linbox/matrix/polynomial-matrix.h>
+#include <linbox/algorithms/polynomial-matrix/polynomial-matrix-domain.h>
+#include <linbox/algorithms/polynomial-matrix/matpoly-mult-naive.h>
+#include <linbox/algorithms/polynomial-matrix/matpoly-mult-kara.h>
+
+using namespace LinBox;
+
+template<typename Rand, typename Matrix>
+void randomMatPol(Rand& r,  Matrix& A){
```

Comment:
This should be just a call to `A.random(r)`. The method exists for `DenseMatrix`. If it does not exist for polynomail matrices is should be added.

Reply:
Right. I've just added the methods to the polynomial matrix classes and correct the program.